### PR TITLE
Binance price oracle

### PR DESCRIPTION
**Pull request by stupid-boar**

### CODE REVIEW

**Comment by stupid-boar** [comment]
```json
{
  "BTC" : {"price":39526,"date":"2022-05-05T07:07:43.253Z"},
  "ETH" : {"price":2927,"date":"2022-05-05T07:07:43.254Z"},
  "WAVES" : {"price":16.55,"date":"2022-05-05T07:07:43.254Z"},
  "EUR" : {"price":1.059,"date":"2022-05-05T07:07:43.254Z"},
  "GBP" : {"price":1.253,"date":"2022-05-05T07:07:43.988Z"},
  "RUB" : {"price":0.015053439710973957,"date":"2022-05-05T07:07:43.991Z"}
}
```

**Review by turbo-stump — APPROVED** [review]

File: tests/unit/oracles/prices/binance.test.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	const oracle = new BinancePriceOracle();
+
+	test('should validate available prices', () => {
+		expectedPrices.map(([asset, expectedAvailable]) => {
+			const available = oracle.isAvailable(asset);
+			expect([asset, available]).toEqual([asset, expectedAvailable]);
+		});
+	});
+
+	it('should successufull fetch price', async () => {
```
Comment (stuffy-idler) [inline tests/unit/oracles/prices/binance.test.ts:28]: Mistype `successfully`

File: tests/unit/oracles/prices/binance.test.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+		fetchMock.mockResponseOnce(JSON.stringify({ lastPrice: '2' }));
+		const data = await oracle.fetchPrice('RUB');
+
+		expect(data.asset).toEqual('RUB');
+		expect(data.price).toEqual(0.5);
+		expect(data.date.getTime()).toBeGreaterThanOrEqual(startTime);
+		expect(data.date.getTime()).toBeLessThanOrEqual(new Date().getTime());
+	});
+
+	it('should fail fetch unavailable price', async () => {
```
Comment (stuffy-idler) [inline tests/unit/oracles/prices/binance.test.ts:50]: `should fail while fetching unavailable price`

File: tsconfig.json [hunk 3]
```diff
@@ -12,6 +12,11 @@
 		"skipLibCheck": true,
 		"sourceMap": true,
 		"strict": true,
-		"target": "es2020"
+		"target": "es2020",
+		"paths": {
+			"src/*": ["src/*"],
```
Comment (stuffy-idler) [inline tsconfig.json:17]: Can it be used in tests instead of `'../../../../src/'`?

**Review by stuffy-idler — APPROVED** [review]

**Pull request merged**